cryostatio/cryostat-agent
# feat(config): add config properties for callback URL components

This work is done to support the [agent auto-configuration](https://github.com/cryostatio/cryostat-operator/issues/784) work in the operator. Specifically, this is needed to support generating wildcard certificates for the agent to use.

To test:
1. Deploy Cryostat using the operator.
2. Deploy the sample app w/ agent.
3. Swap out the default container image with one built using this PR.
4. Create a headless service:
    ```yaml
    apiVersion: v1
    kind: Service
    metadata:
      name: headless
      namespace: cryostat-operator-system
    spec:
      clusterIP: None
      selector:
        app: quarkus-cryostat-agent
      type: ClusterIP
    ```
5. Add a `POD_NAME` environment variable
    ```
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    ```
5.  Add these system properties to `JAVA_OPTS_APPEND`:
    ```
    -Dcryostat.agent.callback.scheme=http
    -Dcryostat.agent.callback.host-name=$(POD_NAME),$(POD_IP)[replace("."\,"-")]
    -Dcryostat.agent.callback.domain-name=headless.cryostat-operator-system.svc
    -Dcryostat.agent.callback.port=9977
    -Dio.cryostat.agent.shaded.org.slf4j.simpleLogger.defaultLogLevel=debug
    ```
6. Observe a lines similar to:
    ```
    2024-11-04 18:41:56:456 +0000 [cryostat-agent-main] DEBUG io.cryostat.agent.ConfigModule - Attempting to resolve quarkus-cryostat-agent-5cd54bbb86-62zsv.headless.cryostat-operator-system.svc
    2024-11-04 18:41:56:546 +0000 [cryostat-agent-main] DEBUG io.cryostat.agent.ConfigModule - Attempting to resolve 10-128-0-67.headless.cryostat-operator-system.svc
    2024-11-04 18:41:56:550 +0000 [cryostat-agent-main] DEBUG io.cryostat.agent.ConfigModule - Resolved 10-128-0-67.headless.cryostat-operator-system.svc to 10.128.0.67
    2024-11-04 18:41:56:552 +0000 [cryostat-agent-main] DEBUG io.cryostat.agent.ConfigModule - Using http://10-128-0-67.headless.cryostat-operator-system.svc:9977 for callback URL
    ```

Fixes: #525
